src/Nest/Modules/SnapshotAndRestore/Repositories/S3Repository.cs

```diff
@@ -7,131 +7,264 @@ public interface IS3Repository : IRepository<IS3RepositorySettings> { }
 
 	public class S3Repository : IS3Repository
 	{
-		public S3Repository(S3RepositorySettings settings)
-		{
-			Settings = settings;
-		}
+		public S3Repository(IS3RepositorySettings settings) => Settings = settings;
 
 		public IS3RepositorySettings Settings { get; set; }
 		public string Type { get; } = "s3";
 	}
 
 	public interface IS3RepositorySettings : IRepositorySettings
 	{
+		/// <summary>
+		/// The name of the bucket to be used for snapshots. This field is required
+		/// </summary>
 		[JsonProperty("bucket")]
 		string Bucket { get; set; }
 
+		/// <summary>
+		/// The region where bucket is located. Defaults to US Standard.
+		/// </summary>
 		[Obsolete("Removed in NEST 6.x.")]
 		[JsonProperty("region")]
 		string Region { get; set; }
 
+		/// <summary>
+		/// The endpoint to the S3 API. Defaults to AWS’s default S3 endpoint. Note that setting a region overrides the endpoint setting.
+		/// </summary>
+		[Obsolete("Removed in NEST 6.x.")]
+		[JsonProperty("endpoint")]
+		string Endpoint { get; set; }
+
+		/// <summary>
+		/// The protocol to use (http or https). Defaults to value of cloud.aws.protocol or cloud.aws.s3.protocol.
+		/// </summary>
+		[Obsolete("Removed in NEST 6.x.")]
+		[JsonProperty("protocol")]
+		string Protocol { get; set; }
+
+		/// <summary>
+		/// Specifies the path within bucket to repository data.
+		/// Defaults to value of repositories.s3.base_path or to root directory if not set.
+		/// </summary>
 		[JsonProperty("base_path")]
 		string BasePath { get; set; }
 
+		/// <summary>
+		/// The access key to use for authentication. Defaults to value of cloud.aws.access_key.
+		/// </summary>
+		[Obsolete("Removed in NEST 6.x. Sensitive settings must be stored in the elasticsearch keystore. See https://www.elastic.co/guide/en/elasticsearch/plugins/current/repository-s3-client.html")]
 		[JsonProperty("access_key")]
 		string AccessKey { get; set; }
 
+		/// <summary>
+		/// The secret key to use for authentication. Defaults to value of cloud.aws.secret_key.
+		/// </summary>
+		[Obsolete("Removed in NEST 6.x. Sensitive settings must be stored in the elasticsearch keystore. See https://www.elastic.co/guide/en/elasticsearch/plugins/current/repository-s3-client.html")]
```

Comment:
Can't be removed in 6.x. See https://github.com/elastic/elasticsearch-net/commit/7d92e6b552d142e14ad0ef04e6668860b29990da. I'm not sure about adding the `Obsolete` attribute in this case. Considering the property will need to remain in 6.x, the `Obsolete` attribute should be applied on the property in 6.x, IMO.

Reply:
Difficult one, yes. it was missed in `6.0` release and was only set as `Obsolete` in the latest `6.x` commit. That said, its it not better to push the user towards avoiding using this property?

Reply:
> That said, its it not better to push the user towards avoiding using this property?

[I don't believe Elasticsearch 5.x supports setting these in elasticsearch keystore](https://www.elastic.co/guide/en/elasticsearch/plugins/5.6/repository-s3-usage.html), _only_ in elasticsearch.yml configuration, so the `Obsolete` message is misleading for 5.x, but I don't think we should apply the attribute to 5.x because it's not helpful - a user can specify access_key, etc. on the request or in configuration in 5.x, and in 6.x, in the keystore only.